src/components/filtermenu/filtermenu.js

```diff
@@ -1,291 +1,302 @@
-        userSettings.setFilter(key, value);
-    }
+    // Genres
+    var genres = [];
+    elems = context.querySelectorAll('.chkGenreFilter');
 
-    function centerFocus(elem, horiz, on) {
-        require(['scrollHelper'], function (scrollHelper) {
-            var fn = on ? 'on' : 'off';
-            scrollHelper.centerFocus[fn](elem, horiz);
-        });
+    for (i = 0, length = elems.length; i < length; i++) {
+        if (elems[i].checked) {
+            genres.push(elems[i].getAttribute('data-filter'));
+        }
     }
-
-    function moveCheckboxFocus(elem, offset) {
-        var parent = dom.parentWithClass(elem, 'checkboxList-verticalwrap');
-        var elems = focusManager.getFocusableElements(parent);
-
-        var index = -1;
-        for (var i = 0, length = elems.length; i < length; i++) {
-            if (elems[i] === elem) {
-                index = i;
-                break;
-            }
+    userSettings.setFilter(settingsKey + '-filter-GenreIds', genres.join(','));
+}
+
+function setBasicFilter(context, key, elem) {
+    var value = elem.checked;
+    value = value ? value : null;
+    userSettings.setFilter(key, value);
+}
+
+function centerFocus(elem, horiz, on) {
+    require(['scrollHelper'], function (scrollHelper) {
+        var fn = on ? 'on' : 'off';
+        scrollHelper.centerFocus[fn](elem, horiz);
+    });
+}
+
+function moveCheckboxFocus(elem, offset) {
+    var parent = dom.parentWithClass(elem, 'checkboxList-verticalwrap');
+    var elems = focusManager.getFocusableElements(parent);
+
+    var index = -1;
+    for (var i = 0, length = elems.length; i < length; i++) {
+        if (elems[i] === elem) {
+            index = i;
+            break;
         }
+    }
 
-        index += offset;
+    index += offset;
 
-        index = Math.min(elems.length - 1, index);
-        index = Math.max(0, index);
+    index = Math.min(elems.length - 1, index);
+    index = Math.max(0, index);
 
-        var newElem = elems[index];
-        if (newElem) {
-            focusManager.focus(newElem);
-        }
+    var newElem = elems[index];
+    if (newElem) {
+        focusManager.focus(newElem);
     }
-
-    function onInputCommand(e) {
-        switch (e.detail.command) {
-            case 'left':
-                moveCheckboxFocus(e.target, -1);
-                e.preventDefault();
-                break;
-            case 'right':
-                moveCheckboxFocus(e.target, 1);
-                e.preventDefault();
-                break;
-            default:
-                break;
-        }
+}
+
+function onInputCommand(e) {
+    switch (e.detail.command) {
+        case 'left':
+            moveCheckboxFocus(e.target, -1);
+            e.preventDefault();
+            break;
+        case 'right':
+            moveCheckboxFocus(e.target, 1);
+            e.preventDefault();
+            break;
+        default:
+            break;
     }
+}
 
-    function FilterMenu() {
+export function FilterMenu() {
```

Comment:
This could be converted to a class.

Reply:
And thanks for contributing to the JF Project!

Reply:
Happy to convert, is `components/appFooter.js` a good template to follow for class conversion @Camc314 ?

Reply:
Yeah it should be. Any question, just ask 😀.